torchrl/envs/libs/pettingzoo.py

```diff
@@ -0,0 +1,872 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+import importlib
+import typing
+from typing import Dict, List, Optional, Tuple, Union
+
+import torch
+from tensordict.tensordict import TensorDictBase
+
+from torchrl.data import (
+    CompositeSpec,
+    DiscreteTensorSpec,
+    OneHotDiscreteTensorSpec,
+    UnboundedContinuousTensorSpec,
+)
+from torchrl.envs.common import _EnvWrapper
+from torchrl.envs.libs.gym import _gym_to_torchrl_spec_transform, set_gym_backend
+from torchrl.envs.utils import _replace_last, check_marl_grouping, MarlGroupMapType
+
+
+_has_pettingzoo = importlib.util.find_spec("pettingzoo") is not None
+
+if typing.TYPE_CHECKING and _has_pettingzoo:
+    import pettingzoo
```

Comment:
can this be a local import?

Reply:
this is used for type checking.
I think it cannot be local if we want the type hints

Reply:
personally I think we can just ditch all type hints but if you want them just do `sutff: "pettingzoo.something"`

Reply:
btw importing a lib for the sake of type hinting is a bit of an overkill no?